### ISSUE TITLE
S2583 false positive when a local variable is initialized through a tuple deconstruction

### ISSUE THREAD

**Issue by volgunin** [SonarSource/sonar-dotnet#3308]
### Description

When a local variable is initialized by a tuple deconstruction, S2583 is raised.

### Repro steps

```C#
        private void Test(bool condition)
        {
            int[] x = null;
            int[] y = null;

            // masks we would like to find
            if (condition)
            {
                (x, y) = F1();
            }

            if (x != null)
            {
                F2();
            }
        }

        static (int[] X, int[] Y) F1() => (new int[10], new int[10]);
        static int F2() => 0;
```
### Expected behavior

No warning.

### Actual behavior

warning S2583: Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Related information

SonarAnalyzer.CSharp.8.6.0.16497
VisualStudio 2019 16.5.4

**Comment by pavel-mikula-sonarsource** [comment]
Hi @volgunin ,

Thank you for reporting this issue. I can confirm this as False Positive behavior. We already have similar case reported in #3239 that has the same root cause. Rules S2583 and S2589 share most of the underlying logic. I'll close this issue since fixing the original one will also fix this problem.
